### DISCUSSION TITLE
Pages are returning build output json instead of html

### DISCUSSION THREAD

**Discussion by JarrodMFlesch** [vercel/community#2677] [Help] [answered]
### Summary

As of recently deploying on Vercel is resulting in some pages returning build output instead of the html for the page for the [payload cms](https://payloadcms.com) website. This happens on different pages sporadically, but normally on _at least_ one of these pages:
- https://payloadcms.com/compare
- https://payloadcms.com/compare/directus (or any of the other compare pages)
- https://payloadcms.com/cloud-pricing
- https://payloadcms.com/for-enterprise

The repo for this is open source, you can find it [here](https://github.com/payloadcms/website).

We have also commented on what seems to be a related NextJS issue [here](https://github.com/vercel/next.js/issues/47805#issuecomment-1568671418).

**NOTE**: This does _**not**_ happen when built and served locally — only when deployed on Vercel.

![CleanShot 2023-05-31 at 08 55 32](https://github.com/vercel/community/assets/30633324/54165d9d-b0f4-43bf-8f38-e16445489a02)

### Example

https://payloadcms.com/compare

### Steps to Reproduce

Visit one of these pages and you will likely see the build output instead of the actual page:
- https://payloadcms.com/compare
- https://payloadcms.com/compare/directus (or any of the other compare pages)
- https://payloadcms.com/cloud-pricing
- https://payloadcms.com/for-enterprise

**Comment by amyegan** [comment]
Hi @JarrodMFlesch. I see what you mean. I reached out to the Next.js team for help with this and am awaiting their feedback 🙇

**Reply by jmikrut** [reply]
Hey @amyegan — this is HUGELY important.

Now, every regular "page" of our site is broken. It was sporadic at first but now appears to have gotten worse. 

Here's the code for the page template which is displaying this error:

https://github.com/payloadcms/website/blob/main/src/app/(pages)/%5B...slug%5D/page.tsx

Note that these exact components exported from this `page.tsx` are also exported and re-used for our homepage, yet on our homepage, this is not an issue.

Here's the homepage:
https://github.com/payloadcms/website/blob/main/src/app/page.tsx

See how the exact components are re-used? 

Strangely enough, _**the homepage does not show this issue.**_ Only pages that use the `[...slug]` page template.

This is hurting our website immensely and we need to expedite this. I believe we are on the Pro tier - is this the official help channel for Pro Vercel customers? Or can we bump this in priority somehow?

**Reply by amyegan** [reply]
I'm sorry I have not been able to identify the cause of this problem yet. I reached out to the engineering team for help and I'll let you know when I have an update from them. 

This is one of our official support channels, but you can also [submit a case](https://vercel.com/help#issues) to the Customer Success Team for a more private line of communication with one of our Customer Success Engineers.

**Reply by jmikrut** [reply]
All good - I will submit now in the hopes that we can get more eyeballs on this issue ASAP. Our team and I are generally pretty self-sufficient, but there are no relevant logs on Vercel to help and this doesn't happen when we deploy elsewhere. So we're pretty much stuck on our end.

Thank you for trying to help here and please continue to do so! We appreciate it!

**Comment by swarnava** [comment]
Hi,

Could disable revalidate/set it to false and let us know if you are still facing this issue?

**Reply by jmikrut** [reply]
Hi there, I've just done this.

It looks like it fixed the issue. So this will tide us over until we get a more permanent fix. Obviously this is not ideal (we did have revalidate set to `600`).

Have you narrowed down the cause of the bug?

**Reply by swarnava** [reply]
Glad to hear that the workaround fixed the issue. Our engineers are looking into it still and unfortunately I don't have a concrete ETA when this will be fixed.

**Reply by jmikrut** [reply]
OK - please keep me posted. For now, happy to see this temporarily resolved as well!

**Comment by ijjk** [comment, marked as answer]
Hi, this has been updated in `v13.4.5` of Next.js please update and give it a try!

**Reply by jmikrut** [reply]
Fixed. THANK YOU @ijjk

**Comment by TrustyTechSG** [comment]
I still randomly see this Error from vercel log with 13.4.13.

`Error: invariant: Expected pageData to be a string for app data request but received undefined. This is a bug in Next.js.`

![image](https://github.com/vercel/community/assets/36871034/67ba7f7d-a15e-464a-9b45-f724193b8abb)

**Comment by JamesSingleton** [comment]
I also have started to see this error randomly for my website backed by Sanity.
![IMG_1863](https://github.com/vercel/community/assets/21000200/b32bc15a-8b3c-4e60-b174-e8333caf0522)

**Comment by IlirEdis** [comment]
Getting this error on my site with embedded Sanity. Having trouble for days now! It breaks my LiveVisualEditing functionality. 
Any fix on this?

Dependencies:

```
"@sanity/client": "^6.14.3",
"@sanity/preview-kit": "^5.0.24",
"@sanity/react-loader": "^1.8.7",
"@sanity/vision": "^3.30.0",
"@sanity/visual-editing": "^1.5.0",
"@vercel/stega": "^0.1.0",
"next-sanity": "^8.0.2",
"next": "14.1.0",

```


<img width="1390" alt="Screenshot 2024-02-28 at 15 11 26" src="https://github.com/vercel/community/assets/128377763/b6ad0729-8ab7-47ef-b4cc-9187316fac0f">

**Comment by ruie** [comment]
I'm also getting this with Sanity
`
"@sanity/client": "^6.4.9",
"next": "^13.4.19",
`

**Reply by IlirEdis** [reply]
I think it got fixed here: https://github.com/vercel/next.js/issues/61377#issuecomment-1996154855
